### PR TITLE
Jf/test model

### PR DESCRIPTION
**Pull request by Jazhyc**
Fixed issues #12 and also #18?

After suffering through 1000s of hyper-parameter testing and a toasty GPU, we have a model that attains ~30% accuracy.
The dimensionality reduction seemingly severely harmed our performance so I added the option to skip it.
Additionally, I added more data-balancing techniques, code to use transformer embeddings and the ability to save our trained model.

We get an accuracy of 29% on the testing set which should be much better than random chance.

### CODE REVIEW

**Comment by nomomon** [comment]
Most epic refactoring I've seen. Good job.

File: create_model.py [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
-def find_best_model():
+def find_best_model(model_type):
     hyperparameters = {
-        "n_dimensions": [10, 25, 50],
-        "n_layers": [1, 2],
-        "n_neurons": [20, 50],
-        "optimizer_type": ["adam", "sgd"],
-        "learning_rate": [0.01, 0.001],
-        "epochs": [5, 10, 20],
+        "dimensionality_reduction": ["none"],
```
Comment (nomomon) [inline create_model.py:7]: Great that you added the skipping some steps such as dim reduction, it's quite interesting that it reduces the performance

File: data/gold/test_word2vec_features.npy
Comment (nomomon) [inline data/gold/test_word2vec_features.npy:1]: It is interesting how this file has been updated. Maybe later we should inspect that.

File: text2emoji/data/embedding_generation.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    # Get last hidden states
+    return tweet_embeddings.pooler_output.detach().cpu()
+
+
+def make_mobert_embeddings(df, name, tokenizer, model):
+
+    tweets = df.text.values.tolist()
+
+    embeddings = []
+    BATCH_SIZE = 256
```
Comment (nomomon) [inline text2emoji/data/embedding_generation.py:86]: This could be a parameter in the function with a default value of 256, maybe someone's computer cannot handle 256 🤷🏻

**Review by nomomon — APPROVED** [review]
